### DISCUSSION TITLE
Improved build notifications

### DISCUSSION THREAD

**Discussion by leerob** [vercel/community#184] [Idea]
### Overview

I want more options to be alerted on my Vercel builds and have improved controls over what gets sent. After pushing a commit, it can be hard to know if the build was successful without closely monitoring the results. We'd like to prevent this workflow:

Refreshing the app → not seeing changes → going to GitHub → seeing deployment failed → clicking and going to Vercel → reading the error ([Source Tweet](https://twitter.com/JokingRajat/status/1481986152651702273))

### Suggestions

- Alerts for build failures
- Additional alerting options (SMS, push notifications)
- Ability to choose which type of events trigger a notification
- In-application notification center

### Resources

- [Original Source](https://twitter.com/shadcn/status/1481662551918538752?s=21)

**Comment by shadcn** [comment]
Thanks for creating this thread @leerob 

I think notifications on failures is a good start (which is what most CI defaults to).

Right now, when you deploy, you can't know if it failed or is still building. 

Adding the failure logs to the email notifications might be useful as well.

**Comment by AnishDe12020** [comment]
Types of notifications can include the following

- Webhooks (posts to a set URL for success failures, URLs can be different for the two)
- Push Notifications, Vercel Notification center (a small one, on the site)
- Emails
- Integrations like Zapier etc. (this is more advanced and may require Vercel to pay but can be done if high demand. note: technically zapier zaps can be triggered with a webhook also)

All the mentioned ones must be customizable to some extent (enable/disable, an option to set only for failed builds (actually, it should default to this), etc.)

**Comment by itsdouges** [comment]
Hi! Thanks for making the thread. Generally the notifications every push are quite overwhelming, I'd love it if it only notified if something failed because right now it actually does the opposite and I need to remember if I don't get a notification something bad happened.

Love your work! :-)

**Comment by iammatthias** [comment]
I'd love to see something akin to Netlify's deploy notifications: https://docs.netlify.com/site-deploys/notifications/

Being able to easily create and configure notification webhooks would be great.

**Comment by leandigital** [comment]
First of all, thanks for all the hard work! With notifications, I'd like to see two options:

1. Push notifications. Something where I can have Vercel to send me an alert on different type of event, configured per project. For example, a notification of a build happening on specific branch, a notification for failed build, etc. These can get pushed to email, Slack, or just a webhook.
2. Subscription to notifications. Something like SNS or Kafka, where I can subscribe to a continuous stream of events and do whatever I need with that. One use case is displaying publishing status in a headless CMS, as otherwise there is no way to know when something is actually published or not.

**Comment by leerob** [comment]
Hey everyone, good news! Our first step towards improving notifications is now live: https://vercel.com/changelog/new-notification-controls-available-in-public-beta

**Comment by steebchen** [comment]
This is great, but it would be so much better if you could configure to only receive notifications (both web and email) for only production deploys. I don't care if a dependabot PR failed deployment, but I do care if a production deploy fails.
